Contest_API.md

```diff
@@ -513,9 +508,9 @@ The endpoints can be categorized into 4 groups as follows:
     awards, commentary;
   - Aggregate data: scoreboard, event-feed.
 
-Metadata is data about the API. These are not included in the event feed and
-are always required to be available. The access endpoint specifies which other
-endpoints are offered by the API. 
+The metadata endpoints contain data about the API, and are the only required API endpoints. 
+They are not included in the event feed. The access endpoint specifies
+which other endpoints are offered by the API. That is, any endpoints and their properties listed in `access` must be provided (possibly with a `null` value when the property is optional), and only these endpoints and properties. 
```

Comment:
When rereading the spec, I now see that the requirement on properties is already specified under `### Table column description` below, and slightly differently, because it says that the properties in `access` *must* be provided, and any that are not there *should* not.
I don't think it's bad to have this repeated, but I think we should put them in sync. I'd be in favor of making it a *must* to have `access` exactly in sync with which properties are there, so only properties listed in `access` may be provided.

Reply:
I definitely agree these places should match, but I'm struggling with the _must_. Compared with other places, having an extra/unexpected attribute doesn't seem like a particularly harmful thing. If you're the kind of client that called `access`, you can easily/safely ignore it if you want to. Yes, you shouldn't do that, but it doesn't really break anything.

@niemela I checked the git blame :) and you wrote the 'should' originally. How do you or @nickygerritsen feel?

Reply:
I’d say I agree: if you can only put fields that are in access that makes extensibility hard right? Say domjudge has a field “name” on account that isn’t in the spec, I don’t necessarily want to put it in access. But then if we add that name field to the spec (which we did right?) are we illegal?

Reply:
I'd say: if you provide an endpoint/property that's outside the spec, then you should still put it in `access`, and then I don't see the problem of making it a *must* to put all your endpoints and properties in `access`. Is there a reason leaving anything that's not defined in the spec out of `access`?

Reply:
So to me `access` is a contract between the server and client about what the server actually provides.

Reply:
I agree all properties (even ones that are custom/beyond the spec) _should_ go in `access`. This lets clients know they exist and may see them, even if they don't know how to handle or will ignore them. I think we should do a separate PR to make this explicit.

My point was that a CCS not putting account name in `access` (or maybe just 'not putting it in _yet_') shouldn't cause harm to any client, so I wasn't convinced it should be a 'must'. To me that property is not spec-compliant, but I would still say the CCS is.

Reply:
I agree with @eldering .

I don't see the benefit of not making this a 'must'.  The spec is actually *very* lenient in what is allowed. Almost everything that's listed is optional, and you may add literally anything else. Requiring that you declare what you have chosen to support from the spec and add to the spec seems like a very reasonable requirement.

Reply:
Ok, other section was updated so that it's a must in both instances.